packages/labs/analyzer/src/lib/program-context.ts

```diff
@@ -0,0 +1,478 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ts from 'typescript';
+import {DiagnosticsError, createDiagnostic} from './errors.js';
+import * as typedoc from 'typedoc';
+import path from 'path';
+import fs from 'fs';
+import {Module, PackageJson, Type, Reference} from './model.js';
+
+type FileCache = ts.MapLike<{version: 0; content?: ts.IScriptSnapshot}>;
```

Comment:
Why not use a real Map?

Reply:
Heh, this was all just based on code cribbed from the TS wiki. Changed to Map.